.github/workflows/ci.yml

```diff
@@ -99,6 +99,103 @@ jobs:
           name: test-results
           path: 'up-cpp/build/test/results/*.xml'
 
+# NOTE: Run dynamic analysis in unit tests
+  valgrind:
+    name: Run up-cpp tests with Valgrind
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - name: Get build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: build-artifacts
+          path: up-cpp/build
+
+      - name: Install Valgrind
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y valgrind
+
+      - name: Run Valgrind Memcheck
+        continue-on-error: true
+        run: |
+          cd up-cpp/build
+          chmod +x bin/*
+          echo "Running Memcheck..."
+          exclude_tests=$(cat valgrind_exclude_tests.txt | tr '\n' '|' | sed 's/|$//')
+          echo "Excluding tests: $exclude_tests"
+          set +e
+          valgrind --leak-check=full \
+                   --show-leak-kinds=all \
+                   --track-origins=yes \
+                   --trace-children=yes \
+                   --verbose \
+                   --log-file=valgrind-memcheck.log \
+                   --gen-suppressions=all \
+                   ctest -E "$exclude_tests"
+
+          exit_code=$?
+          set -e
+
+          if [ $exit_code -ne 0 ]; then
+            touch memcheck_failed_output.log
+            echo "Some tests failed. Capturing the logs..."
+            cat Testing/Temporary/LastTest.log | tee -a memcheck_failed_output.log
+          else
+            echo "No tests failed."
+            touch memcheck_failed_output.log
+          fi
```

Comment:
Does this output the test log from the test that failed?

Reply:
It will have some summary from the test that failed